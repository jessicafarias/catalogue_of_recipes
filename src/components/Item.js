import PropTypes from 'prop-types';
import '../styles/slider.scss';
import Style from 'style-it';

const Item = ({ url, title, order }) => {
  let stringclass = '';

  if (order === 0) {
    stringclass = 'active';
  }
  return Style.it(`
  #uberuns:after {
    content:'';
    background: url("${url}");
    background-attachment:fixed;
    background-size:cover;
  }`,
    <div className={`carousel-item ${stringclass}`}>

      <div className="image" style={{ backgroundImage: `url(${url})` }} />
      <div className="section" id="uberuns" data={`url(${url})`}>
        <p className="text_slider">
          {title}
        </p>
        <p className="span_title pb-2">By Themealdb</p>
        <div className="row m-2 p-2">
          <div className="col-4 d-flex justify-content-center align-items-center">
            <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="#e9d061" className="bi bi-star-fill" viewBox="0 0 16 16">
              <path d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.283.95l-3.523 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z" />
            </svg>
            <p className="tag_slider">FAVORITE</p>
          </div>
          <div className="col-4 d-flex justify-content-center align-items-center">
            <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="#e9d061" className="bi bi-heart-fill" viewBox="0 0 16 16">
              <path fillRule="evenodd" d="M8 1.314C12.438-3.248 23.534 4.735 8 15-7.534 4.736 3.562-3.248 8 1.314z" />
            </svg>
            <p className="tag_slider">RECOMMEND</p>
          </div>
          <div className="col-4 d-flex justify-content-center align-items-center">
            <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="#e9d061" className="bi bi-share-fill" viewBox="0 0 16 16">
              <path d="M11 2.5a2.5 2.5 0 1 1 .603 1.628l-6.718 3.12a2.499 2.499 0 0 1 0 1.504l6.718 3.12a2.5 2.5 0 1 1-.488.876l-6.718-3.12a2.5 2.5 0 1 1 0-3.256l6.718-3.12A2.5 2.5 0 0 1 11 2.5z" />
            </svg>
            <p className="tag_slider">SHARE</p>
          </div>
        </div>
      </div>
    </div>);
};

Item.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  order: PropTypes.number.isRequired,
};

export default Item;
