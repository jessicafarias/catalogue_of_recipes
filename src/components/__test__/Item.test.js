import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import '@testing-library/jest-dom';
import TestRenderer from 'react-test-renderer'; // ES6
import Item from '../Item';

describe('Display Item.js', () => {
  it('renders correctly', () => {
    const tree = TestRenderer.create(<BrowserRouter><Item /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
