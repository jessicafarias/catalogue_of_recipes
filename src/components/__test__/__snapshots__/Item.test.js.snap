// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Item.js renders correctly 1`] = `
<div
  className="carousel-item active s1193061191"
>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".s1193061191  #uberuns:after {
 content:''; background: url(\\"https://images.pexels.com/photos/4193850/pexels-photo-4193850.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940\\"); background-attachment:fixed; background-size:cover; }
",
      }
    }
    type="text/css"
  />
  <div
    className="image"
    style={
      Object {
        "backgroundImage": "url(https://images.pexels.com/photos/4193850/pexels-photo-4193850.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940)",
      }
    }
  />
  <div
    className="section"
    data="url(https://images.pexels.com/photos/4193850/pexels-photo-4193850.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940)"
    id="uberuns"
  >
    <p
      className="text_slider"
    >
      No found
    </p>
    <p
      className="span_title pb-2"
    >
      By Themealdb
    </p>
    <div
      className="row m-2 p-2"
    >
      <div
        className="col-4 d-flex justify-content-center align-items-center"
      >
        <svg
          className="bi bi-star-fill"
          fill="#e9d061"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.283.95l-3.523 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
          />
        </svg>
        <p
          className="tag_slider"
        >
          FAVORITE
        </p>
      </div>
      <div
        className="col-4 d-flex justify-content-center align-items-center"
      >
        <svg
          className="bi bi-heart-fill"
          fill="#e9d061"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 1.314C12.438-3.248 23.534 4.735 8 15-7.534 4.736 3.562-3.248 8 1.314z"
            fillRule="evenodd"
          />
        </svg>
        <p
          className="tag_slider"
        >
          RECOMMEND
        </p>
      </div>
      <div
        className="col-4 d-flex justify-content-center align-items-center"
      >
        <svg
          className="bi bi-share-fill"
          fill="#e9d061"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 2.5a2.5 2.5 0 1 1 .603 1.628l-6.718 3.12a2.499 2.499 0 0 1 0 1.504l6.718 3.12a2.5 2.5 0 1 1-.488.876l-6.718-3.12a2.5 2.5 0 1 1 0-3.256l6.718-3.12A2.5 2.5 0 0 1 11 2.5z"
          />
        </svg>
        <p
          className="tag_slider"
        >
          SHARE
        </p>
      </div>
    </div>
  </div>
</div>
`;
